cdap-ui/app/directives/log-viewer/log-viewer.js

```diff
@@ -360,33 +356,33 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi, LOGVIEWERSTORE_
 
   const pollForNewLogs = () => {
     pollPromise = dataSrc.poll({
-      _cdapPath: '/namespaces/' + this.namespaceId + '/apps/' + this.appId + '/' + this.programType + '/' + this.programId + '/runs/' + this.runId + '/logs?format=json&start=' + this.startTimeSec,
+      _cdapPath: '/namespaces/' + this.namespaceId + '/apps/' + this.appId + '/' + this.programType + '/' + this.programId + '/runs/' + this.runId + '/logs/next?format=json&max=100&fromOffset=' + this.fromOffset,
       method: 'GET'
     },
     (res) => {
       //We have recieved more logs, append to current dataset
-      if(res.length > this.data.length){
-        res = res.slice(this.data.length, res.length);
-
-        angular.forEach(res, (element, index) => {
-          if(res[index].log.logLevel === 'WARN'){
-            this.warningCount++;
-          } else if(res[index].log.logLevel === 'ERROR'){
-            this.errorCount++;
-          }
+    if(res.length > 0){
+      this.fromOffset = res[res.length-1].offset;
+
+      angular.forEach(res, (element, index) => {
+        if(res[index].log.logLevel === 'WARN'){
+          this.warningCount++;
+        } else if(res[index].log.logLevel === 'ERROR'){
+          this.errorCount++;
+        }
 
-          //Format dates properly for rendering and computing
-          let formattedDate = new Date(res[index].log.timestamp);
-          res[index].log.timestamp = formattedDate;
-          res[index].log.displayTime = ((formattedDate.getMonth() + 1) + '/' + formattedDate.getDate() + '/' + formattedDate.getFullYear() + ' ' + formattedDate.getHours() + ':' + ((formattedDate.getMinutes()<10) ? '0'+formattedDate.getMinutes() : formattedDate.getMinutes()) + ':' + formattedDate.getSeconds());
-          res[index].log.stackTrace = res[index].log.stackTrace.trim();
-        });
+        //Format dates properly for rendering and computing
+        let formattedDate = new Date(res[index].log.timestamp);
+        res[index].log.timestamp = formattedDate;
+        res[index].log.displayTime = ((formattedDate.getMonth() + 1) + '/' + formattedDate.getDate() + '/' + formattedDate.getFullYear() + ' ' + formattedDate.getHours() + ':' + ((formattedDate.getMinutes()<10) ? '0'+formattedDate.getMinutes() : formattedDate.getMinutes()) + ':' + formattedDate.getSeconds());
+        res[index].log.stackTrace = res[index].log.stackTrace.trim();
+      });
 
-        this.data = this.data.concat(res);
-        this.renderData(true);
-      }
+      this.data = this.data.concat(res);
+      this.renderData();
+    }
 
-      if(this.displayData.length > this.viewLimit){
+      if(this.displayData.length >= 100){
```

Comment:
We should probably have 100 as constant. It would be helpful if we want to change this in the future (or better yet give this number as an option in UI)